### ISSUE TITLE
.NET Automation API uses incorrect git argument (`--remote-git-username`)

### ISSUE THREAD

**Issue by PhilBroderickCrezco** [pulumi/pulumi-dotnet#185]
### What happened?

Following the remote deployment example here: https://github.com/pulumi/automation-api-examples/tree/main/dotnet/RemoteDeployment, using `RemoteGitProgramArgs` to authenticate with a private GitHub organization repository, I end up with the following error:

```
Pulumi.Automation.Commands.Exceptions.CommandException: code: 1
stdout: 
stderr: Error: unknown flag: --remote-git-username
Usage:
  pulumi preview [url] [flags]

Aliases:
  preview, pre

Flags:
  -c, --config stringArray                            Config to use during the preview
      --config-file string                            Use the configuration values in the specified file rather than detecting the file name
      --config-path                                   Config keys contain a path to a property in a map or list to set
  -d, --debug                                         Print detailed debugging output during resource operations
      --diff                                          Display operation as a rich diff showing the overall change
      --event-log string                              Log events to a file at this path
      --expect-no-changes                             Return an error if any changes are proposed by this preview
  -h, --help                                          help for preview
  -j, --json                                          Serialize the preview diffs, operations, and overall output as JSON
  -m, --message string                                Optional message to associate with the preview operation
  -p, --parallel int                                  Allow P resource operations to run in parallel at once (1 for no parallelism). Defaults to unbounded. (default 2147483647)
      --policy-pack strings                           Run one or more policy packs as part of this update
      --policy-pack-config strings                    Path to JSON file containing the config for the policy pack of the corresponding "--policy-pack" flag
  -r, --refresh string[="true"]                       Refresh the state of the stack's resources before this update
      --remote                                        [EXPERIMENTAL] Run the operation remotely
      --remote-env --remote-env FOO=bar               [EXPERIMENTAL] Environment variables to use in the remote operation of the form NAME=value (e.g. --remote-env FOO=bar)
      --remote-env-secret --remote-env FOO=secret     [EXPERIMENTAL] Environment variables with secret values to use in the remote operation of the form NAME=secretvalue (e.g. --remote-env FOO=secret)
      --remote-git-auth-access-token string           [EXPERIMENTAL] Git personal access token
      --remote-git-auth-password string               [EXPERIMENTAL] Git password; for use with username or with an SSH private key
      --remote-git-auth-ssh-private-key string        [EXPERIMENTAL] Git SSH private key; use --remote-git-auth-password for the password, if needed
      --remote-git-auth-ssh-private-key-path string   [EXPERIMENTAL] Git SSH private key path; use --remote-git-auth-password for the password, if needed
      --remote-git-auth-username string               [EXPERIMENTAL] Git username
      --remote-git-branch string                      [EXPERIMENTAL] Git branch to deploy; this is mutually exclusive with --remote-git-branch; either value needs to be specified
      --remote-git-commit string                      [EXPERIMENTAL] Git commit hash of the commit to deploy (if used, HEAD will be in detached mode); this is mutually exclusive with --remote-git-branch; either value needs to be specified
      --remote-git-repo-dir string                    [EXPERIMENTAL] The directory to work from in the project's source repository where Pulumi.yaml is located; used when Pulumi.yaml is not in the project source root
      --remote-pre-run-command stringArray            [EXPERIMENTAL] Commands to run before the remote operation
      --remote-skip-install-dependencies              [EXPERIMENTAL] Whether to skip the default dependency installation step
      --replace stringArray                           Specify resources to replace. Multiple resources can be specified using --replace urn1 --replace urn2
      --save-plan string                              [EXPERIMENTAL] Save the operations proposed by the preview to a plan file at the given path
      --show-config                                   Show configuration keys and variables
      --show-reads                                    Show resources that are being read in, alongside those being managed directly in the stack
      --show-replacement-steps                        Show detailed resource replacement creates and deletes instead of a single step
      --show-sames                                    Show resources that needn't be updated because they haven't changed, alongside those that do
      --show-secrets false                            Emit secrets in plaintext in the plan file. Defaults to false
  -s, --stack string                                  The name of the stack to operate on. Defaults to the current stack
      --suppress-outputs                              Suppress display of stack outputs (in case they contain sensitive values)
      --suppress-permalink string[="false"]           Suppress display of the state permalink
  -t, --target stringArray                            Specify a single resource URN to update. Other resources will not be updated. Multiple resources can be specified using --target urn1 --target urn2
      --target-dependents                             Allows updating of dependent targets discovered but not specified in --target list
      --target-replace stringArray                    Specify a single resource URN to replace. Other resources will not be updated. Shorthand for --target urn --replace urn.

Global Flags:
      --color string                 Colorize output. Choices are: always, never, raw, auto (default "auto")
  -C, --cwd string                   Run pulumi as if it had been started in another directory
      --disable-integrity-checking   Disable integrity checking of checkpoint files
  -e, --emoji                        Enable emojis in the output
      --logflow                      Flow log settings to child processes (like plugins)
      --logtostderr                  Log to stderr instead of to files
      --memprofilerate int           Enable more precise (and expensive) memory allocation profiles by setting runtime.MemProfileRate
      --non-interactive              Disable interactive mode for all commands
      --profiling string             Emit CPU and memory profiles and an execution trace to '[filename].[pid].{cpu,mem,trace}', respectively
      --tracing file:                Emit tracing to the specified endpoint. Use the file: scheme to write tracing data to a local file
      --tracing-header string        Include the tracing header with the given contents.
  -v, --verbose int                  Enable verbose logging (e.g., v=3); anything >3 is very verbose

An error occurred: unknown flag: --remote-git-username


   at Pulumi.Automation.Commands.LocalPulumiCmd.RunAsyncInner(IList`1 args, String workingDir, IDictionary`2 additionalEnv, Action`1 onStandardOutput, Action`1 onStandardError, EventLogFile eventLogFile, CancellationToken cancellationToken)
   at Pulumi.Automation.Commands.LocalPulumiCmd.RunAsync(IList`1 args, String workingDir, IDictionary`2 additionalEnv, Action`1 onStandardOutput, Action`1 onStandardError, Action`1 onEngineEvent, CancellationToken cancellationToken)
   at Pulumi.Automation.Commands.LocalPulumiCmd.RunAsync(IList`1 args, String workingDir, IDictionary`2 additionalEnv, Action`1 onStandardOutput, Action`1 onStandardError, Action`1 onEngineEvent, CancellationToken cancellationToken)
   at Pulumi.Automation.Workspace.RunStackCommandAsync(String stackName, IList`1 args, Action`1 onStandardOutput, Action`1 onStandardError, Action`1 onEngineEvent, CancellationToken cancellationToken)
   at Pulumi.Automation.WorkspaceStack.RunCommandAsync(IList`1 args, Action`1 onStandardOutput, Action`1 onStandardError, Action`1 onEngineEvent, CancellationToken cancellationToken)
   at Pulumi.Automation.WorkspaceStack.PreviewAsync(PreviewOptions options, CancellationToken cancellationToken)
   at Pulumi.Automation.WorkspaceStack.PreviewAsync(PreviewOptions options, CancellationToken cancellationToken)
   at Program.<Main>$(String[] args) in C:\dev\Crezco-App\infrastructure\Automation\Program.cs:line 27
```

I notice that the `--remote-git-auth-username` is marked as `[EXPERIMENTAL]` but so are the other git related args.

### Example

Sensitive information redacted:

```csharp
var stackArgs = new RemoteGitProgramArgs(projectName, <ORG_GIT_REPO>)
{
    Branch = "refs/heads/master",
    ProjectPath = project,
    Auth = new RemoteGitAuthArgs
    {
        Username = <Username>,
        PersonalAccessToken = <PersonalAccessToken>
    }
};

var stack = await RemoteWorkspace.SelectStackAsync(stackArgs);
```

### Output of `pulumi about`

CLI                              
Version      3.78.0              
Go Version   go1.20.7            
Go Compiler  gc                  
                                 
Host                             
OS       Microsoft Windows 11 Pro
Version  10.0.22621 Build 22621  
Arch     x86_64   

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
> However it seems you don't allow PRs from the community? 

We definitely do, if you fork the repo and push a change to your copy you should be able to raise a PR against this repo.

If you want to give that a go happy to let you take the credit for the fix, else it's a small enough change we can get a fix and release done this week ourselves.

**Comment by PhilBroderickCrezco** [comment]
Thanks @Frassle. I was having some unrelated issues that I correlated with being unable to open a PR against this repo. I've not resolved those issues and raised a new PR to fix this issue 😃

**Comment by Frassle** [comment]
Fixed in #186, thanks for the PR.
